torch/utils/data/sampler.py

```diff
@@ -104,10 +104,14 @@ def num_samples(self):
 
     def __iter__(self):
         n = len(self.data_source)
+        if self.generator is None:
+            self.generator = torch.default_generator
         if self.replacement:
-            rand_tensor = torch.randint(high=n, size=(self.num_samples,), dtype=torch.int64, generator=self.generator)
-            return iter(rand_tensor.tolist())
-        return iter(torch.randperm(n, generator=self.generator).tolist())
+            for _ in range(self.num_samples // n):
```

Comment:
Isn't `self.num_samples` (number of samples to draw) generally going to be smaller or equal to `n` (dataset size)? That is, won't `self.num_samples // n` < 1? Should it be the other way around?

Reply:
I interpreted the original issue as ``n`` is much larger than ``num_samples``, and that's why ``torch.randint()`` would take much space. Otherwise the space won't be larger than ``torch.randperm()`` for non-replacement.
Maybe I interpreted it the wrong way. Can someone clarify on this?

Reply:
@choidongyeon is right. Use some fixed constant for the size at each time, e.g., 32 or something, and don't assume you get zero remainder.